### ISSUE TITLE
Rails 6.1 deprecation warning

### ISSUE THREAD

**Issue by lubosch** [mailgun/mailgun-ruby#215]
this issue: https://github.com/rails/rails/issues/36546 is caused by this gem because of 

```ruby
# /lib/railgun/railtie.rb
ActionMailer::Base.add_delivery_method :mailgun, Railgun::Mailer
``` 
https://github.com/rails/rails/issues/36546#issuecomment-763792736

and it can be fixed I guess by https://github.com/rails/rails/issues/36546#issuecomment-764865398
```ruby
ActiveSupport.on_load(:action_mailer) do
  add_delivery_method :mailgun, Railgun::Mailer
end
```

**Comment by cgunther** [comment]
#162 is an attempt to address this.

**Comment by lubosch** [comment]
official response from mailgun support:
> We do not have an exact date as this is something that is being worked on. Our goal is to have it done in roughly the next few months

So either don't use mailgun, this gem, fix it yourself or don't upgrade rails next few months ...

**Comment by zokioki** [comment]
Looks like a fix for this was just introduced in `v1.2.4`:
https://github.com/mailgun/mailgun-ruby/blob/v1.2.4/lib/railgun/railtie.rb#L5-L8.
